### PR TITLE
Fix vendor dashboard product edit page variation products 

### PR DESCRIPTION
**Pull request by Aunshon**
### All Submissions:

* [ ] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [ ] My code satisfies feature requirements
* [ ] My code is tested
* [ ] My code passes the PHPCS tests
* [ ] My code has proper inline documentation
* [ ] I've included related pull request(s) (optional)
* [ ] I've included developer documentation (optional)
* [ ] I've added proper labels to this pull request

### Related Pull Request(s)

* https://github.com/getdokan/plugin-vendor-dashboard/pull/119
* https://github.com/getdokan/dokan-pro/pull/3530

### Closes 

* Closes #

### How to test the changes in this Pull Request:

* Steps or issue link

### Changelog entry

*Title*

Detailed Description of the pull request. What was previous behaviour
and what will be changed in this PR.

### Before Changes

Describe the issue before changes with screenshots(s).

### After Changes

Describe the issue after changes with screenshot(s).

### Feature Video (optional)

Link of detailed video if this PR is for a feature.

### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.

### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.

* [ ] Correct — Does the change do what it’s supposed to? ie: code 100% fulfilling the requirements?
* [ ] Secure — Would a nefarious party find some way to exploit this change? ie: everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities?
* [ ] Readable — Will your future self be able to understand this change months down the road?
* [ ] Elegant — Does the change fit aesthetically within the overall style and architecture?

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method for retrieving linked products dynamically based on search criteria and user IDs.
	- Enhanced API response to return detailed product attribute information instead of a simple success/failure status.

- **Bug Fixes**
	- Improved error handling in product attribute management by ensuring attribute checks are performed on arrays.

- **Refactor**
	- Optimized product search and filtering logic for better performance and maintainability.

### CODE REVIEW

File: includes/Product/Manager.php [hunk 1]
```diff
@@ -714,4 +714,42 @@ public function top_rated( $args = [] ) {
...
+     * @param integer               $limit
+     *
+     * @return WC_Product[]
+     */
+    public function get_linked_products( $term = '', $user_ids = false, $exclude = null, $included_id = null, $limit = 0 ) {
+        $term     = ! empty( $term ) ? sanitize_text_field( wp_unslash( $term ) ) : '';
+        $user_ids = ! empty( $user_ids ) ? array_filter( array_map( 'absint', (array) wp_unslash( $user_ids ) ) ) : false;
+
+        if ( empty( $term ) ) {
+            wp_die();
```
Comment (mrabbani) [inline includes/Product/Manager.php:736]: Pls through exception instead of `die`. 

**Pls never use `die`.**

File: includes/Product/Manager.php [hunk 2]
```diff
@@ -714,4 +714,42 @@ public function top_rated( $args = [] ) {
...
+
+    /**
+     * Returns linked products.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @param string                $term
+     * @param boolean|integer|array $user_ids
+     * @param integer               $exclude
+     * @param integer               $included_id
```
Comment (mrabbani) [inline includes/Product/Manager.php:726]: Exclude and include params should be array instead of single interger.

**Review by mrabbani — CHANGES_REQUESTED** [review]

**Comment by StalinDurjo** [comment]
**Issue: For variable products, the min-max quantity vales cannot be saved**
**Description**: Vendor has created a variable product and tries to set the min-max quantity limit for an individual variation. However, setting the values and saving does not actually set any value.

Please refer to the video attached
https://jam.dev/c/c52da8b5-bee9-43f6-8447-1d2e4f0e513c

@Aunshon bhai

File: includes/Ajax.php [hunk 3]
```diff
@@ -651,22 +651,15 @@ public function json_search_product() {
...
-        }
-
-        if ( ! empty( $_GET['include'] ) ) {
-            $ids = array_intersect( $ids, (array) sanitize_text_field( wp_unslash( $_GET['include'] ) ) );
-        }
-
-        if ( ! empty( $_GET['limit'] ) ) {
-            $ids = array_slice( $ids, 0, absint( $_GET['limit'] ) );
-        }
+        $product_objects = dokan()->product->get_linked_products(
```
Comment (mrabbani) [inline includes/Ajax.php:654]: Does the `get_linked_products` method exist?

**Review by mrabbani — APPROVED** [review]
